pymedphys/_dicom/anonymise.py

```diff
@@ -350,14 +362,23 @@ def anonymise_directory(
         with caution, since unrecognised tags may contain identifying
         information.
     """
-    dicom_filepaths = glob(str(dicom_dirpath) + "/**/*.dcm", recursive=True)
+    dicom_dirpath = str(dicom_dirpath)
+
+    dicom_filepaths = glob(dicom_dirpath + "/**/*.dcm", recursive=True)
     failing_filepaths = []
     # errors = []
 
     for dicom_filepath in dicom_filepaths:
+        if output_dirpath is not None:
+            relative_path = os.path.relpath(dicom_filepath, start=dicom_dirpath)
+            output_filepath = os.path.join(output_dirpath, relative_path)
+        else:
+            output_filepath = None
+
         anonymise_file(
             dicom_filepath,
-            delete_original_file=False,
```

Comment:
@Matthew-Jennings This was a logic error here. The delete_original file parameter for the directory function was being ignored.